tests/integration/test_merge_tree_s3_restore/test.py

```diff
@@ -103,7 +105,7 @@ def create_table(
         ORDER BY (dt, id)
         SETTINGS
             storage_policy='s3',
-            old_parts_lifetime=600,
+            old_parts_lifetime=1,
```

Comment:
For being able to wait until covered parts and empty part are removed.
A few lines I added a call wait_for_delete_empty_parts.

Reply:
I spent more time to be able answer this question more proper.
Lets consider such actions
0 -- initial parts: all_1_1_0 (active)
1 -- some mutation, parts are: all_1_1_0 (outdated) all_1_1_1 (active)
after this operation all_1_1_0 is deleted only after old_parts_lifetime sec.
2 -- drop part all, parts are
        Now: all_1_1_0 (outdated) all_1_1_1 (outdated) all_1_1_2 (active) --- after this operation all_1_1_1 could be deleted instantly (as we agreed) but all_1_1_2 will be wait until all_1_1_0 is deleted.
        Before: all_1_1_0 (outdated) -- just drop all_1_1_1 and go on, part is not in active set.

So, something similar is happeing in test_restore_to_detached test. It freeze empty active part all_1_1_2. This part is cover the one with data which has been detached before freeze. The part with data is lost.
To prevent this, I modified test to just wait until new empty part is gone. But cleaning this part depends on covered outdated part after mutation. old_parts_lifetime need for clear old parts from mutation.